### DISCUSSION TITLE
[Вопрос] Перенос ключей из LXC контейнера

### DISCUSSION THREAD

**Discussion by TheSainEyereg** [xtrime-ru/antizapret-vpn-docker#56] [Q&A] [answered]
Мне нужно, чтобы старые конфиги, которые я раздал людям и настроил на устройствах продолжили работать при переходе с LXC на Docker. Сервер тот же, поэтому единственное что нужно -- подкорректировать порт (что тривиально) и перенести ключи чтобы конфиги оставались валидными. Как я могу это сделать?

**Comment by xtrime-ru** [comment]
Запуститите клиент, что бы сгенерировались ключи. А потом нужно подменить соответствующие файлы в папке ./keys . В LXD контейнере серверные ключи в  `/etc/openvpn`, а клиентские ключи там в `/root/easy-rsa-ipsec/pki/private/`

После подмены файлов перезапустить контейнер и проверить, что в обновленных ovpn файлах ключи совпадают с нужными.

**Reply by TheSainEyereg** [reply]
Не вышло. Перенес всё содержимое LXC из `/root/easy-rsa-ipsec/easyrsa3/pki/private/` в `./keys/client/keys/`, а содержимое `/etc/openvpn/server/` в `./keys/server/` и при последующем запуске ключи в конечных ovpn файлах оказались отличными от тех, что изначально выдавал антизапрет под LXC. 

При этом если глянуть на сами файлы которые "нужно подменить", то можно заметить, что, как минимум, расширения у некоторых из них разные.
![image](https://github.com/user-attachments/assets/ac206d67-709b-4b36-be59-ffeec05b5b26)
![image](https://github.com/user-attachments/assets/559b3c88-41d6-429d-bdbe-56679b4fff9a)

К примеру: появились снова файлы `antizapret-client.crt` и `antizapret-tls-crypt.key`. Их нет в изначальном LXC контейнере и они создаются даже если их удалить. Может быть нужно ещё где-то конфиги подправить?

**Reply by xtrime-ru** [reply, marked as answer]
antizapret-tls-crypt.key это новый файл, он испльзуется если включена опция OPENVPN_TLS_CRYPT=1. По умолчанию он не будет использоваться. 

Вот наш скрипт: https://github.com/xtrime-ru/antizapret-vpn-docker/blob/master/rootfs/root/openvpn/generate.sh#L58

Скопируйте из старого контейнера /root/easyrsa/pki в новый контейнер, потом удалите папку ./keys на хосте и перезапустите контейнер.

**Reply by flameshikari** [reply]
И не забудь про переменную `OPENVPN_CBC_CIPHERS=1`, поскольку LXC-контейнер использовал старые CBC-шифры.

**Reply by TheSainEyereg** [reply]
Сработало. Пришлось правда в названом generate.sh закомментировать build_pki. После этого перенес файлы в контейнер, перезапустил его и ключи в файлах стали совпадать, спасибо
